### PR TITLE
fix: add per-field length validation to MetadataOverrides (#294)

### LINKED ISSUE #294: [Tech Debt] High - Book metadata string fields have no server-side length caps
Book metadata string fields — `title`, `description`, author names, tag names, and series names — pass through the API layer with no server-side length validation. SQLite `TEXT` columns are unbounded by default, and while `MetadataOverrides::validate()` is called on the override endpoint, it does not enforce character-count limits on these fields.

This is a HIGH severity gap because an authenticated user (or an admin with library write access) can submit arbitrarily large payloads for individual text fields. A 50 MB `description` or a 1 MB `title` will be stored and served back to every client loading that book's detail page, consuming unbounded memory in the response serializer and potentially causing client-side rendering failures. The existing 1 MiB global body limit (`axum::extract::DefaultBodyLimit`) provides a ceiling, but no per-field limit prevents a single field from consuming the entire budget.

Issue #276 (username length validation, now open) covers the auth registration path. This issue covers the complementary gap in the book metadata write path. Neither the `POST /api/ebooks/{uuid}/overrides` endpoint (`server/src/backend/overrides.rs:30`) nor `MetadataOverrides::validate()` (`db/src/metadata_overrides.rs`) enforce per-field length constraints on `title`, `description`, `authors`, `tags`, or `series`.

Add explicit length caps in `MetadataOverrides::validate()` — e.g., title ≤ 500 chars, description ≤ 50,000 chars, author/tag/series name ≤ 250 chars. Return a `400 Bad Request` with a descriptive field-level error message. Mirror the same caps in any future RPC server function that writes overrides (`frontend/src/rpc.rs`). Consider adding a `#[test]` for the boundary values alongside existing `validate()` tests.

### PR DESCRIPTION
**Pull request by seamus-sloan**
## Summary

- Replaced the single `MAX_SCALAR_CHARS = 4096` constant with three purpose-scoped public constants that match natural per-field budgets:
  - `MetadataOverrides::TITLE_MAX_LEN = 500` chars
  - `MetadataOverrides::DESCRIPTION_MAX_LEN = 50_000` chars
  - `MetadataOverrides::NAME_MAX_LEN = 250` chars (publisher, series, language, series_index, creator names)
  - `MetadataOverrides::TAG_MAX_LEN = 128` chars (tag/subject strings — see note below)
- Updated `validate()` in `shared/src/lib.rs` to use the new per-field constants
- Added a REST integration test in `server/src/backend/overrides.rs`: `api_post_overrides_rejects_over_length_title_with_400` — verifies that an over-limit title returns 400 and does not write an override row
- Updated all unit tests in `shared/src/lib.rs` to use the new constants and added boundary tests for description and series

## Why

The previous implementation had a single `MAX_SCALAR_CHARS = 4096` cap for every short field (title, publisher, etc.) and `MAX_DESCRIPTION_CHARS = 256 * 1024` for the description. While length validation was present, the limits were too permissive: a 4096-char title uses ~16 KB at 4 bytes/char, and a 262,144-char description uses ~1 MB — well above the safe per-field budgets described in the issue. The new values cap the worst case at ~2 KB for title and ~200 KB for description.

`validate()` is already called at both ingress points:
- `server/src/backend/overrides.rs` — REST endpoint (mobile client)
- `frontend/src/rpc.rs` — server function (web client)

No DB migration needed — SQLite TEXT columns remain unbounded; the cap is enforced purely at the application layer.

## Tag cap: pre-existing `MAX_SUBJECT_CHARS = 128` preserved as `TAG_MAX_LEN = 128`

An adversarial reviewer noted that an earlier revision of this PR silently loosened the tag cap from 128 → 250 by folding tags into `NAME_MAX_LEN`. The pre-existing `MAX_SUBJECT_CHARS = 128` has been preserved as `TAG_MAX_LEN = 128`. `NAME_MAX_LEN = 250` applies only to author/series/publisher names as specified in issue #294. Tags are typically short controlled-vocabulary terms, so the tighter cap is intentional.

## Tests added

- `validate_rejects_title_exceeding_length_cap` — updated to check the 500-char cap (was 4096)
- `validate_accepts_title_at_length_cap` — boundary at exactly 500 chars
- `validate_rejects_description_exceeding_length_cap` — new, checks 50,000-char cap
- `validate_accepts_description_at_length_cap` — new, boundary at exactly 50,000 chars
- `validate_rejects_series_name_exceeding_length_cap` — new, checks 250-char NAME_MAX_LEN
- `validate_measures_title_cap_in_chars_not_bytes` — updated (was `validate_measures_scalar_cap_in_chars_not_bytes`)
- `validate_rejects_over_long_tag` — tag strings checked against TAG_MAX_LEN=128; over-limit boundary is 129 chars
- `validate_accepts_tag_at_length_cap` — new, boundary at exactly 128 chars
- `api_post_overrides_rejects_over_length_title_with_400` — new REST integration test

## Pre-existing notes

None — all checks were green on main before branching. `cargo clippy --all-targets -- -D warnings` and `cargo test -p omnibus-db && cargo test -p omnibus` pass clean.

Closes #294

https://claude.ai/code/session_01PDc9MZzA951sCBXnL7LBF4

### CODE REVIEW

File: shared/src/lib.rs [hunk 1]
```diff
@@ -158,55 +158,51 @@ pub struct MetadataOverrides {
...
             if creators.len() > Self::MAX_CREATORS {
                 return Err(format!("too many creators (max {})", Self::MAX_CREATORS));
             }
             for c in creators {
-                if c.name.chars().count() > Self::MAX_CREATOR_CHARS {
+                if c.name.chars().count() > Self::NAME_MAX_LEN {
                     return Err(format!(
                         "creator name exceeds {} characters",
-                        Self::MAX_CREATOR_CHARS
+                        Self::NAME_MAX_LEN
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply shared/src/lib.rs:208]: Fixed in f80fd77 — added checks for `c.role` and `c.file_as` against `NAME_MAX_LEN` inside the `creators` loop in `validate()`. Also added two new tests: `validate_rejects_over_long_creator_role` and `validate_rejects_over_long_creator_file_as`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: shared/src/lib.rs [hunk 2]
```diff
@@ -158,55 +158,51 @@ pub struct MetadataOverrides {
...
 
 impl MetadataOverrides {
-    /// Maximum length (in Unicode scalar values, i.e. `chars`) for scalar
-    /// string fields (title, publisher, etc.).
-    const MAX_SCALAR_CHARS: usize = 4096;
-    /// Maximum length (in chars) for the description field.
-    const MAX_DESCRIPTION_CHARS: usize = 256 * 1024;
+    /// Maximum length (in Unicode scalar values, i.e. `chars`) for the `title`
+    /// field.
+    pub const TITLE_MAX_LEN: usize = 500;
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply shared/src/lib.rs]: Fixed in f80fd77 — collapsed to a single-line summary: `/// Maximum title length in Unicode scalar values (chars).`

---
_Generated by [Claude Code](https://claude.ai/code)_

File: shared/src/lib.rs [hunk 3]
```diff
@@ -158,55 +158,51 @@ pub struct MetadataOverrides {
...
+    /// Maximum length (in chars) for the `description` field.
+    pub const DESCRIPTION_MAX_LEN: usize = 50_000;
+    /// Maximum length (in chars) for single-line name fields: `publisher`,
+    /// `published`, `language`, `series`, `series_index`, creator names, and
+    /// tag strings.
+    pub const NAME_MAX_LEN: usize = 250;
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply shared/src/lib.rs]: Fixed in f80fd77 — collapsed to a single-line summary: `/// Maximum length in chars for single-line name fields: publisher, series, language, and creator names.`

---
_Generated by [Claude Code](https://claude.ai/code)_
